### PR TITLE
chore: improve UserUpdateUsername returns

### PR DESCRIPTION
**Pull request by vindard**
## Description

Improve the responses from the `UserUpdateUsername` mutation:
- on success, includes the username instead of `null` in the response
- on "is immutable" edge case, still returns a user object as feedback (didn't before this)

### CODE REVIEW

File: src/graphql/root/mutation/user-update-username.ts [hunk 1]
```diff
@@ -30,7 +31,9 @@ const UserUpdateUsernameMutation = GT.Field({
 
     if (result instanceof Error) {
       const appErr = mapError(result)
-      return { errors: [{ message: appErr.message || appErr.name }] }
+      return result instanceof UsernameIsImmutableError
+        ? { errors: [{ message: appErr.message || appErr.name }], user: domainUser }
+        : { errors: [{ message: appErr.message || appErr.name }] }
```
Comment (dolcalmi) [inline src/graphql/root/mutation/user-update-username.ts:36]: this is not necessary `message: appErr.message || appErr.name` please use only `[mapError(result)]`

File: src/graphql/types/object/graphql-user.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ const GraphQLUser = GT.Object({
       type: Username,
       description: "Optional immutable user friendly identifier.",
       resolve: async (source, args, { domainAccount }) => {
-        return domainAccount?.username
+        return domainAccount?.username || source.username
```
Comment (dolcalmi) [inline src/graphql/types/object/graphql-user.ts:38]: why is this needed?

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: src/graphql/root/mutation/user-update-username.ts (hunk 1 above)
Comment (vindard) [reply src/graphql/root/mutation/user-update-username.ts:36]: I tested this, see: https://github.com/GaloyMoney/galoy/pull/1125#discussion_r813879550

**Review by bodymindarts — APPROVED** [review]
